segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/PravegaRequestProcessor.java

```diff
@@ -607,6 +615,16 @@ public void mergeSegmentsBatch(WireCommands.MergeSegmentsBatch mergeSegments) {
                });
     }
 
+    private long getIndexSegmentMaxEventSize(String segmentName) {
```

Comment:
This method is calling join on the future which blocks the thead. This is not allowed on the Segment Store. So inline this into the callers and then chain the `processAppend` call on the future.

Reply:
Addressed the same